examples/gin/main.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/airbrake/gobrake/v5"
+	ginbrake "github.com/airbrake/gobrake/v5/gin"
+	"github.com/gin-gonic/gin"
+)
+
+var ProjectId int64 = 105138                               // Replace it with your project Id
+var ProjectKey string = "fd04e13d806a90f96614ad8e529b2822" // Replace it with your project key
+
+// Define models
+
+type Date struct {
+	Date int64 // the server date
+}
+
+type WeatherInfo struct {
+	Lat            float32       `json:"lat,omitempty"`
+	Lon            float32       `json:"lon,omitempty"`
+	Timezone       string        `json:"timezone,omitempty"`
+	TimezoneOffset float32       `json:"timezone_offset,omitempty"`
+	Current        interface{}   `json:"current,omitempty"`
+	Minutely       []interface{} `json:"minutely,omitempty"`
+	Hourly         []interface{} `json:"hourly,omitempty"`
+	Daily          []interface{} `json:"daily,omitempty"`
+}
+
+var availableLocations = []string{
+	"austin",
+	"pune",
+	"santabarbara",
+}
+
+var notifier = gobrake.NewNotifierWithOptions(&gobrake.NotifierOptions{
+	ProjectId:   ProjectId,    // use it as an environment variable
+	ProjectKey:  ProjectKey,   // use it as an environment variable
+	Environment: "production", // use it as an environment variable
+})
+
+func main() {
+
+	api := gin.Default()
+
+	// Initialise middlewares
+	api.Use(ginbrake.New(api, notifier))
+	api.Use(TokenAuthMiddleware())
+
+	// Initialise routes
+	api.GET("/date", getDate)
+	api.GET("/locations", getLocations)
+	api.GET("/weather/:location", getWeather)
+
+	// Initialise application
+	api.Run(":3000")
+}
+
+// Define handlers for routes
+
+func getDate(c *gin.Context) {
+	fmt.Println(c.Request.Header.Get("api-key"))
+	now := time.Now()
+	secs := now.Unix()
+	c.JSON(http.StatusOK, gin.H{"date": secs})
+}
+
+func getLocations(c *gin.Context) {
+	c.JSON(http.StatusOK, availableLocations)
+}
+
+func getWeather(c *gin.Context) {
+	var weatherInfo WeatherInfo
+	location := strings.TrimSpace(c.Param("location")) // It lets you allow whitespaces but case-sensitive
+	if !contains(availableLocations, location) {
+		c.JSON(http.StatusNotFound, weatherInfo)
+		return
+	}
+	weatherInfoFile := location + ".json"
+	weatherInfoFilePath := filepath.Join("asset", weatherInfoFile)
```

Comment:
```suggestion
	weatherInfoFilePath := filepath.Join("assets", weatherInfoFile)
```

This is a more common name than `asset`.